tests/test_tokenization_common.py

```diff
@@ -1398,6 +1398,65 @@ def test_right_and_left_padding(self):
                 assert sequence_length == padded_sequence_left_length
                 assert encoded_sequence == padded_sequence_left
 
+    def test_right_and_left_truncation(self):
+        tokenizers = self.get_tokenizers(do_lower_case=False)
+        for tokenizer in tokenizers:
+            with self.subTest(f"{tokenizer.__class__.__name__}"):
+                sequence = "This is a test sequence"
+
+                # RIGHT PADDING - Check that it correctly pads when a maximum length is specified along with the padding flag set to True
+                truncation_size = 3
+                tokenizer.truncation_side = "right"
+                encoded_sequence = tokenizer.encode(sequence, add_special_tokens=False)
+                sequence_length = len(encoded_sequence)
+                truncated_sequence = tokenizer.encode(
+                    sequence, max_length=sequence_length - truncation_size, truncation=True, add_special_tokens=False
+                )
+                truncated_sequence_length = len(truncated_sequence)
+                assert sequence_length == truncated_sequence_length + truncation_size
```

Comment:
Please use `self.assertEqual` in tests (here and all occurrences below)

Reply:
Done ! (There are still quite a bit of old `assert` in the rest of the file, should I do it in a separate PR ?

Reply:
If you have time, that would be great!

Reply:
https://github.com/huggingface/transformers/pull/15012